xkcd.py

```diff
@@ -1,22 +1,71 @@
 from Module import Command
+
+from bs4 import BeautifulSoup
+from urllib import request
+from datetime import datetime
+
 import string
 
-module_name = "xkcd"
+xkcd_comics = {
+    "updated" : datetime(1900, 1, 1),
+    "comics"  : []
+}
 
 def command_xkcdrandomnumber(cmd, bot, args, msg, event):
     return "[4 // Chosen by fair dice roll. Guaranteed to be random.](http://xkcd.com/221/)"
 
-
 def command_xkcd(cmd, bot, args, msg, event):
+    global xkcd_comics
+
     if len(args) < 1:
         return "Not enough arguments."
+
+    # If the first parameter is an integer, assume it's the XKCD strip number.
     try:
-        id_ = int(args[0])
+        id = int(args[0])
+        return "http://xkcd.com/{0}".format(id)
     except:
-        return "Invalid arguments."
-    return "http://xkcd.com/%i/" % id_
+        pass
+
+    # TODO It would be nice if we could also pull the alt-text
+    if xkcd_comics["updated"] < datetime.today():
+        page = request.urlopen("http://xkcd.com/archive/")
+        soup = BeautifulSoup(page, 'html.parser')
+        page.close()
+
+        xkcd_comics["comics"] = []
+        div = soup.find(id="middleContainer")
+
+        for link in div.find_all("a"):
+            for string in link.stripped_strings:
+                xkcd_comics["comics"].append({ "id" : link["href"][1:-1], "title": string })
+
+        xkcd_comics["updated"] = datetime.today()
+
+    results = []
+    filter = " ".join(args[0:]).lower()
+    for comic in xkcd_comics["comics"]:
+        if comic["title"].lower().find(filter) > -1:
+            results.append(comic)
+
+    if len(results) == 0:
+        return "No matching comic found."
+
+    if len(results) == 1:
+        return "http://xkcd.com/{0}".format(results[0]["id"])
+
+    if len(results) > 5:
+        return "Too many matching comics, be more specific please!"
+
+    bot.room.send_message(":{0} More than one comic found; try `%%xkcd <#>` with one of these numbers:".format(event.message.id))
+    for xkcd in results:
+        bot.room.send_message("{0}: {1}".format(xkcd["id"], xkcd["title"]))
```

Comment:
@ArtOfCode- Yeah, that's what Mike told me in chat yesterday, so then I proposed to multiline messages, because sending so many messages slows down the bot because of rate limiting.

Additionally, that piece of code wouldn't even get executed because if would get executed *after* the `return` statement /cc @KutuluMike